packages/material-ui-styles/src/propsToClassKeys/propsToClassKeys.test.js

```diff
@@ -0,0 +1,34 @@
+import { expect } from 'chai';
+import propsToClassKeys from './propsToClassKeys';
+
+describe('propsToClassKeys', () => {
+  it('should return the variant value as string', () => {
+    expect(propsToClassKeys({ variant: 'custom' })).to.equal('custom');
+  });
+
+  it('should combine the variant with other props', () => {
+    expect(propsToClassKeys({ variant: 'custom', size: 'large' })).to.equal('customSizeLarge');
```

Comment:
Now I understand. It wasn't clear to me whether we sort by keys or values. It makes more sense to me to sort by keys instead, no? Otherwise we `{ variant: 'large', size: 'medium' }` and `{ variant: 'medium', size: 'large' }` will generate the same key. That might be a bit overly pessimistic but for readability I would expect sorting by keys anyway so that it's immediately obvious which value a prop has.

Reply:
The values are sorted by key, the thing is that the variant is always first if it is defined (not sure if it is the best approach, but it was mainly done because that's how most of the classkeys are implemented already), so I didn't want to introduce too many breaking changes (there is a special case for the color prop too to not have the key first).. 

So basically, variant value is always first if exists, then all other props are added by alphabetical order.

Reply:
Missed that the key is part of the classkey